cpp/include/kvikio/utils.hpp

```diff
@@ -330,4 +347,29 @@ struct libkvikio_domain {
   } while (0)
 #endif
 
+/**
+ * @brief Convenience macro for generating an NVTX marker in the `libkvikio` domain to annotate a
+ * certain time point.
+ *
+ * Takes two arguments (message, payload). Use this macro to annotate asynchronous I/O operations,
+ * where the payload refers to the I/O size.
+ *
+ * Example:
+ * ```
+ * std::future<void> some_function(){
+ *     size_t io_size{2077};
+ *     KVIKIO_NVTX_MARKER("I/O operation", io_size);
+ *     perform_async_io_operation(io_size);
+ *     ...
+ * }
+ * ```
+ */
+#ifdef KVIKIO_CUDA_FOUND
```

Comment:
NVTX can be used without CUDA in host-only code. We can disable this if we don't care about profiling no-CUDA code paths, though.

Reply:
I think the guard is needed, we only build with NVTX support when `KVIKIO_CUDA_FOUND` is defined: https://github.com/rapidsai/kvikio/blob/branch-24.12/cpp/CMakeLists.txt#L92

Reply:
Okay, sounds good. It is _possible_ to use NVTX without CUDA, but I'm fine with this outcome given the lack of CMake support for that.